src/librustc_middle/ty/instance.rs

```diff
@@ -336,6 +351,24 @@ impl<'tcx> Instance<'tcx> {
         tcx.resolve_instance(tcx.erase_regions(&param_env.and((def_id, substs))))
     }
 
+    // This should be kept up to date with `resolve`.
+    pub fn resolve_const_arg(
```

Comment:
Oops, I missed this one, could this be `resolve_opt_const_arg`?
Maybe `resolve` could call into this with `ty::WithOptConstParam::unknown(def_id)`, to avoid duplication?